File: typescript/7kyu/Mumbling.ts
export const accum = (s: string): string => {
  const stringArr = s.toLowerCase().split("");
  const result: string[] = [];
  for (let i = 0; i < stringArr.length; i++) {
    i === 0
      ? result.push(`${stringArr[i].toUpperCase()}`)
      : result.push(`${stringArr[i].toUpperCase()}${stringArr[i].repeat(i)}`);
  }
  return result.join("-");
};

File: python/7kyu/Mumbling.py
def accum(s):
    string_list = list(s.lower())
    result = []
    for i in range(len(string_list)):
        if i == 0:
            result.append(string_list[i].upper())
        else:
            result.append(string_list[i].upper() + (string_list[i] *  i))
    
    return "-".join(result)